web/assets/main.js

```diff
@@ -15,3 +15,25 @@ GOVUKFrontend.Tabs.prototype.setup = () => {};
 
 GOVUKFrontend.initAll();
 MOJFrontend.initAll();
+
+if (window.self !== window.top) {
+  document.body.className += " app-!-embedded";
+
+  const success = document.querySelector(".moj-banner--success");
+  if (success) {
+    window.top.postMessage(
+      "form-done",
+      `${window.location.protocol}//${window.location.host}`
+    );
+  }
+
+  document.querySelectorAll("[data-app-iframe-cancel]").forEach((el) => {
```

Comment:
How many of these elements are you expecting? Another way might be to add a document.click handler and filter events by data attribute to keep the number of handlers down in the app. This may be overkill though example here https://github.com/ministryofjustice/opg-digideps/blob/main/client/assets/javascripts/modules/googleAnalyticsEvents.js#L3

Reply:
Really just one, but this will nicely handle the 0 element case if we don't have a cancel button